pages/checkout/googlepay.vue

```diff
@@ -0,0 +1,133 @@
+<template>
+  <div>
+    <div id="payment-page">
+      <div class="container">
+        <div class="payment-container">
+          <div id="component-container" class="payment"></div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup>
+
+import { ref, onMounted } from 'vue';
+import { AdyenCheckout, GooglePay } from '@adyen/adyen-web';
+
+// Reactive references
+const sessionId = ref('');
+const redirectResult = ref('');
+
+// Function to invoke server-side API
+async function sendPostRequest(url, data) {
```

Comment:
The `sendPostRequest` has been duplicated many times on top-level. Let's extract this in a separate helper/util function

_Reason:_ Does javascript not take the last function and overrides the other functions? E.g. when we start combining components on one page, it will take the last function call. This will introduce some unintended side-effects. 

`sendPostRequest(..)` in ideal.vue <-- Modifying this function won't have any effect
`sendPostRequest(..)` in googlepay.vue <-- as this one is the last javascript function executed*

Let's fix this by extracting these functions and reuse them:

- [ ] sendPostRequest
- [ ] handleServerResponse
- [ ] createAdyenCheckout
- [ ] startCheckout
- [ ] finalizeCheckout
- [ ] onMounted

Reply:
On second thought: if the split is intended, we keep it, let's do the `sendPostRequest`-only